lib/entities/team-membership.js

```diff
@@ -114,3 +110,28 @@ export function wrapTeamMembershipCollection (http, data, teamId) {
   teamMemberships.items = teamMemberships.items.map((entity) => wrapTeamMembership(http, entity, teamId))
   return freezeSys(teamMemberships)
 }
+
+/**
+ * @private
+ * @param {Object} http - HTTP client instance
+ * @param {Object} data - Raw teamMembership data
+ * @return {TeamMembership} Wrapped TeamMembership data
+ */
+export function wrapOrganizationTeamMembership (http, data) {
+  const teamMembership = toPlainObject(cloneDeep(data))
+  return freezeSys(teamMembership)
+}
+
+/**
+ * @private
+ * @param {Object} http - HTTP client instance
+ * @param {Object} data - Raw teamMembership data
+ * @return {TeamMembershipCollection} Wrapped TeamMembershipCollection data
+ */
+export function wrapOrganizationTeamMembershipCollection (http, data) {
```

Comment:
Can you explain to me again why the teamMemberships returned from `/organization/<orgId>/team_memberships` shouldn't have update/delete, but the teamMemberships returned from `organization/<orgId>/teams/<teamId>/team_memberships` should?

Reply:
The endpoint `organization/orgid/team_memberships` only has [GET] implemented and there is no story to implement show, update, delete. To manage particular teamMemberships one should use `getTeamMembership(teamId, teamMembershipId)`. https://app.stoplight.io/wk/cduzvQ7Fbbsj2mRk2/7knfjqR4Y5n8oj7hq/6vEAKeemm9QaKW2Pq/design/endpoints/phK5PdTGRPfLukiJq

Reply:
I get that you cant PUT or DEL to `/organization/orgid/team_memberships`, but wont the objects returned from that API include a teamID?

From the perspective of the user of the SDK, if what I get back is a collection of team membership entities (regardless of the path from which I requested it), I would expect to be able to update/delete each of them. Plus, if the objects in the `items` returned from `/organization/orgid/team_memberships` have a teamID attached to them, we can easily wrap them using the existing method.

this is the power of the SDK :) it doesnt have to directly mimic the API design, it should be straight forward and easy to use. For me, it's confusing that there would be two types of teamMembership entities with different APIs defined on them.

Reply:
the only reason that I would see to keep it with a different fetching method and a different wrap method would be:
the actual items returned from `/organization/orgid/team_memberships` are so different, contain different information, do not include the team id.

Reply:
hmmm, yes I see what you mean by that

Reply:
ok let me try to update and see if the tests pass and cover the cases! thank you!

Reply:
A compromise I could live with:

The method to fetch is different, but it's still wrapped with the same wrapper function

Reply:
its ok, I thought the issue was the other way around :D i didnt understand the concern from before